docs/guides/sre/configuring-ssl.md

```diff
@@ -0,0 +1,440 @@
+openssl req -newkey rsa:2048 -nodes -subj "/CN=chnode1" -addext "subjectAltName = DNS:chnode1.marsnet.local,IP:192.168.1.221" -keyout chnode1.key -out chnode1.csr
+openssl req -newkey rsa:2048 -nodes -subj "/CN=chnode2" -addext "subjectAltName = DNS:chnode2.marsnet.local,IP:192.168.1.222" -keyout chnode2.key -out chnode2.csr
+openssl req -newkey rsa:2048 -nodes -subj "/CN=chnode3" -addext "subjectAltName = DNS:chnode3.marsnet.local,IP:192.168.1.223" -keyout chnode3.key -out chnode3.csr
+```
+
+6. Using the CSR and CA, create new certificate and key pairs
+```bash
+openssl x509 -req -in chnode1.csr -out chnode1.crt -CAcreateserial -CA marsnet_ca.crt -CAkey marsnet_ca.key -days 365
+openssl x509 -req -in chnode2.csr -out chnode2.crt -CAcreateserial -CA marsnet_ca.crt -CAkey marsnet_ca.key -days 365
+openssl x509 -req -in chnode3.csr -out chnode3.crt -CAcreateserial -CA marsnet_ca.crt -CAkey marsnet_ca.key -days 365
+```
+
+7. Verify that certs for subject and issuer.
+```bash
+openssl x509 -in chnode1.crt -text -noout
+```
+
+8. Check that the new certificates verify against the CA cert
+```bash
+openssl verify -CAfile marsnet_ca.crt chnode1.crt
+chnode1.crt: OK
+```
+
+
+## 3. Create and Configure a directory to store certificates and keys.
+:::note
+This must be done on each node. Use approriate certificates and keys on each host.
+:::
+1. Create a folder in a directory accessible by ClickHouse in each node, recommended is the default configuration directory (e.g. /etc/clickhouse-server)
+```bash
+mkdir /etc/clickhouse-server/certs
+```
+
+2. Copy the CA certificate, node certifiate and key corresponding to each node to the new certs directory
+
+3. Update owner and permissions to allow ClickHouse to read the certificates
+```bash
+chown clickhouse:clickhouse -R /etc/clickhouse-server/certs
+chmod 666 /etc/clickhouse-server/certs/*
+chmod 755 /etc/clickhouse-server/certs
+ll /etc/clickhouse-server/certs
+total 20
+drw-r--r-- 2 clickhouse clickhouse 4096 Apr 12 20:23 ./
+drwx------ 5 clickhouse clickhouse 4096 Apr 12 20:23 ../
+-rw-r--r-- 1 clickhouse clickhouse  997 Apr 12 20:22 chnode1.crt
+-rw-r--r-- 1 clickhouse clickhouse 1708 Apr 12 20:22 chnode1.key
+-rw-r--r-- 1 clickhouse clickhouse 1131 Apr 12 20:23 marsnet_ca.crt
+```
+
+## 4. Configure the environment with basic clusters using ClickHouse Keeper 
+
+For this deployment environment, the following ClickHouse Keeper settings are used in each node. Each server will have its own `<server_id>`. e.g. `<server_id>1</server_id>` for node `chnode1`, etc.
+
+:::note
+Recommended port is `9281` for ClickHouse Keeper. However, the port is configurable and can be set if this port is in use already by another application in the environment.
+The Zookeeper standard ports can also be used, `2181` for non-SSL and `2281` for SSL.
+:::
+
+For full options: https://clickhouse.com/docs/en/operations/clickhouse-keeper/
+
+1. add the following inside the `<clickhouse>` tag in clickhouse-server `config.xml`
```

Comment:
We probably generally recommend putting extra config in the conf.d subdirectory instead of directly in config.xml (I think that's the right place).  That way they can keep this config separate

Reply:
I can add extra steps, was just thinking of minimal steps possible to get started.

Reply:
Understood, I'm just not sure where to draw the line.  I guess I don't want to have an example of modifying config.xml directly if we're going to recommend using "included" files as a general practice.

Reply:
Maybe just a note someplace that separate xml (or yaml) files can be added into the appropriate directory instead of modifying config.xml

Reply:
that sounds good. I'll add a ":::note" thing that states best practice/recommendations.